reel-picker/src/main/java/com/innowisegroup/reelpicker/datetime/LocalDate.kt

```diff
@@ -2,11 +2,36 @@ package com.innowisegroup.reelpicker.datetime
 
 import com.innowisegroup.reelpicker.extension.formattedMonth
 import com.innowisegroup.reelpicker.extension.isLeapYear
+import com.innowisegroup.reelpicker.extension.isWithinMinMaxRange
 import com.innowisegroup.reelpicker.extension.requireNonNull
 import java.io.Serializable
 import java.util.*
 
-class LocalDate(val day: Int, val month: Int, val year: Int) : Serializable {
+class LocalDate(day: Int, month: Int, year: Int) : Serializable {
+
+    internal var day = day
```

Comment:
Can we avoid "var" for day, month and year. And use "val" instead 
Let's use by lazy delegate here. The same for month and year

    internal val day by lazy {
        validateDay(day)
        day
    }